concepts/architecture/index.md

```diff
@@ -289,6 +289,11 @@ The Autonity Stabilization Contract implements logic for a liquidator to:
 
 To learn more about the concept see [Auton Stability Mechanism (ASM)](/concepts/asm/).
 
+### Protocol contract upgrade
+
+Autonity protocol contracts are upgradable by governance calling the [`upgrade()`](/reference/api/aut/op-prot/#upgrade) function on the `upgradeManagerContract` to provide new contract bytecode and abi for a designated protocol contract address.
+
+Upgrade functions by replacing contract logic for a designated contract address. The contract address and replacement contract code is passed in as parameters to the `upgrade()` function call. The block [`finalize()`](/reference/api/aut/op-prot/#finalize) function checks if a contract upgrade is available. The protocol will then update the contract code of the autonity contract during the block finalization phase. Contract code is replaced in the EVM and existing contract state is maintained.
```

Comment:
The upgrade at finalize is how the native upgrade mechanism for the AC works (it is still there). 

However with the new UpgradeManager contract the bytecode substitution is immediate (as soon as the tx is executed).

Reply:
@lorenzo-dev1 - edited in commit f5420c5cb309f92ea91a7ee169ebc06c47ede543

Please review 👍